### ISSUE TITLE
Broken for threads

### ISSUE THREAD

**Issue by mileslucas** [timholy/ProgressMeter.jl#143]
The new Julia 1.3 `Threads.@threads` macro does not work with `@showprogress`

MWE:
```julia
using ProgressMeter
a = []
@showprogress Threads.@threads for i in 1:1000
    push!(a, rand())
end
```

Python's tqdm figured out a way to show progress on multiple processes by showing a stack of progress bars for each process. Perhaps a similar interface could work here?

**Comment by ianshmean** [comment]
The non-macro Progress methods do work with threads (there are some examples in the readme) but it’s true that the macros haven’t been adapted for threads yet.

**Comment by MarcMush** [comment]
fixed by #284
